pkg/api/store/globalrolebindings/globalrolebinding_store.go

```diff
@@ -8,21 +8,48 @@ import (
 
 const GrbVersion = "field.cattle.io/grbUpgrade"
 
-func Wrap(store types.Store, grbLister v3.GlobalRoleBindingLister) types.Store {
+func Wrap(store types.Store, grLister v3.GlobalRoleLister) types.Store {
 	return &grbStore{
-		Store:     store,
-		grbLister: grbLister,
+		Store:    store,
+		grLister: grLister,
 	}
 }
 
 type grbStore struct {
 	types.Store
-	grbLister v3.GlobalRoleBindingLister
+
+	grLister v3.GlobalRoleLister
 }
 
 func (s *grbStore) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
 
 	values.PutValue(data, "true", "metadata", "annotations", GrbVersion)
 
+	err := s.addOwnerReference(data)
+	if err != nil {
+		return nil, err
+	}
+
 	return s.Store.Create(apiContext, schema, data)
 }
+
+// addOwnerReference ensures that a globalRolebinding will be deleted when the role it references
+// is deleted
+func (s *grbStore) addOwnerReference(data map[string]interface{}) error {
+	globalRoleName, _ := data["globalRoleId"].(string)
+
+	globalRole, err := s.grLister.Get("", globalRoleName)
+	if err != nil {
+		return err
+	}
+
+	ownerReference := map[string]interface{}{
+		"apiVersion": globalRole.APIVersion,
+		"kind":       globalRole.Kind,
+		"name":       globalRole.Name,
+		"uid":        globalRole.UID,
+	}
+	values.PutValue(data, []map[string]interface{}{ownerReference}, "ownerReferences")
```

Comment:
`ownerReferences` is not a top level attribute, this needs to be under metadata and it's a list of objects.

Reply:
After some discovery it appears things in the store should follow the view of being a client object - meaning everything at the top level. There are issues with doing metadata at both levels which causes things to be dropped once the obj reaches the toInternal stage.